configuration/pih/liquibase/sql/create_functions.sql

```diff
@@ -2009,7 +2109,8 @@ BEGIN
   from patient_state ps
   inner join program_workflow_state pws on ps.state = pws.program_workflow_state_id and program_workflow_id =_program_workflow_id  
   where ps.patient_program_id = _patient_program_id
-  and ps.end_date is null;
+  and ps.end_date is null
```

Comment:
The null end date is always an interesting question with programs.  Since all states are ended (I think) when a program enrollment is ended, this will end up showing no current program state.  But if you want to show the most recent program state for a given workflow in their enrollment, then you'll need to tweak this.  This can be a common issue if, say, you have a workflow for "Treatment Status" that have values that include terminal states like "Died".  In these cases, "Died" will never show up as the current program state.

Reply:
This makes sense.  I didn't actually change that but will make a fix for what you describe.

Reply:
@mseaton, edited this function in the PR.  When you have a chance, take a look and let me know if this is what you had in mind.